flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -329,6 +330,21 @@ object UserDefinedFunctionUtils {
   }
 
   /**
+    * Returns the value (boolean) of AggregateFunction#requiresOver() that indicates if this
+    * AggregateFunction can only be used for over window aggregate. If method requiresOver is not
+    * provided, return false as default value.
+    */
+  def getRequiresOverConfig(aggregateFunction: AggregateFunction[_, _]): Boolean = {
+    try {
+      val method: Method = aggregateFunction.getClass.getMethod("requiresOver")
```

Comment:
Yes, I think we should check the return type of `requiresOver` as well , in order to give an explicit exception.